src/ode_solver/runge_kutta_base.cpp

```diff
@@ -0,0 +1,78 @@
+#include "runge_kutta_base.h"
+
+namespace PHiLiP {
+namespace ODE {
+
+template<int dim, typename real, int n_rk_stages, typename MeshType>
+RungeKuttaBase<dim, real, n_rk_stages, MeshType>::RungeKuttaBase(std::shared_ptr< DGBase<dim, real, MeshType> > dg_input,
+            std::shared_ptr<EmptyRRKBase<dim,real,MeshType>> RRK_object_input)
+            : ODESolverBase<dim,real,MeshType>(dg_input)
+            , relaxation_runge_kutta(RRK_object_input)
+            , solver(dg_input)
+{}            
+
+
+template<int dim, typename real, int n_rk_stages, typename MeshType>
+void RungeKuttaBase<dim, real, n_rk_stages, MeshType>::step_in_time(real dt, const bool pseudotime)
+{
+    this->original_time_step = dt;
+    this->solution_update = this->dg->solution; //storing u_n
+    for (int i = 0; i < n_rk_stages; ++i){
+        this->calculate_stages(i, dt, pseudotime); // u_n + dt * sum(a_ij * k_j) <explicit> + dt * a_ii * u^(i) <implicit>
+        this->apply_limiter(); // Might need to change for LSRK
+        this->obtain_stage(i, dt); //rk_stage[i] = IMM*RHS = F(u_n + dt*sum(a_ij*k_j))
+    }
+    dt = this->adjust_time_step(dt);
+    this->sum_stages(dt, pseudotime); // u_np1 = u_n + dt* sum(k_i * b_i)
+    this->dg->solution = this->solution_update; 
+     // Calculate numerical entropy with FR correction. Does nothing if use has not selected param.
+    this->FR_entropy_contribution_RRK_solver = relaxation_runge_kutta->compute_FR_entropy_contribution(dt, this->dg, this->rk_stage, true);
+    this->apply_limiter();
+    ++(this->current_iteration);
+    this->current_time += dt;
+}
+
+template<int dim, typename real, int n_rk_stages, typename MeshType>
+void RungeKuttaBase<dim, real, n_rk_stages, MeshType>::allocate_ode_system()
+{
+    this->pcout << "Allocating ODE system..." << std::flush;
+    this->solution_update.reinit(this->dg->right_hand_side);
+
+    this->pcout << std::endl;
+    
+    this->rk_stage.resize(n_rk_stages);
```

Comment:
I think this should be done in the derived classes, as the LSRK minimizes storage requirements to need a number smaller than `n_rk_stages`. Therefore, only allocating `solution_update` is needed in this base class.

Reply:
I've changed the LSRK allocation to include the following line
`// Clear the rk_stage object for memory optimization`
`this->rk_stage.clear();`

This will clear the rk_stage vector to minimize the storage requirements. I have done this in this way as every other RK method needs these rk_stages. If we implent more methods that don't need `this->rk_stage`, then we can update this in the future.

Let me know if you would still like to move this into the class specific allocation functions.